pulp_docker/app/tasks/recursive_add.py

```diff
@@ -0,0 +1,58 @@
+from pulpcore.plugin.models import Repository, RepositoryVersion
+
+from pulp_docker.app.models import Manifest, ManifestBlob, ManifestTag, MEDIA_TYPE
+
+
+def recursive_add_content(repository_pk, content_units):
+    """
+    Create a new repository version by recursively adding content.
+
+    For each unit that is specified, we also need to add related content. For example, if a
+    manifest-list is specified, we need to add all referenced manifests, and all blobs referenced
+    by those manifests.
+
+    Args:
+        repository_pk (int): The primary key for a Repository for which a new Repository Version
+            should be created.
+        content_units (list): List of PKs for :class:`~pulpcore.app.models.Content` that
+            should be added to the previous Repository Version for this Repository.
+    """
+    repository = Repository.objects.get(pk=repository_pk)
+
+    tags_to_add = ManifestTag.objects.filter(
+        pk__in=content_units,
+    )
+
+    manifest_lists_to_add = Manifest.objects.filter(
+        pk__in=content_units,
+        media_type=MEDIA_TYPE.MANIFEST_LIST
+    ) | Manifest.objects.filter(
+        pk__in=tags_to_add.values_list('tagged_manifest', flat=True),
+        media_type=MEDIA_TYPE.MANIFEST_LIST,
+    )
+
+    manifests_to_add = Manifest.objects.filter(
+        pk__in=content_units,
+        media_type__in=[MEDIA_TYPE.MANIFEST_V1, MEDIA_TYPE.MANIFEST_V1_SIGNED,
```

Comment:
we can probably use negation here media_type __not=manifest_list

Reply:
not would need to include all the blob types too, so I figured this was easier

Reply:
Im surprised, but that didnt work. 
"Unsupported lookup \'not\' for CharField or join on the field not permitted.",

Reply:
interesting, seems like for this we need to implement a custom lookup?
let's leave the code then as is for now